keanu-project/src/main/java/io/improbable/keanu/util/io/VertexDotLabel.java

```diff
@@ -32,23 +33,33 @@ public void setValue(String value) {
     public String inDotFormat() {
         // Output value if value is set, but also add some descriptive info for non-constant vertices.
         if (!value.isEmpty()) {
-            String dotLabel = vertex.getId().hashCode() + DOT_LABEL_OPENING + value;
-            if (!(vertex instanceof ConstantVertex)) {
-                dotLabel += " (" + getDescriptiveInfo() + ")";
-            }
-            return dotLabel + DOT_LABEL_CLOSING;
+            StringBuilder dotLabel = new StringBuilder();
+            dotLabel.append(vertex.getId().hashCode()).append(DOT_LABEL_OPENING).append(value);
+            getDescriptiveInfoForVertexWithValue().ifPresent(info -> dotLabel.append(" (").append(info).append(")"));
+            return dotLabel.append(DOT_LABEL_CLOSING).toString();
         }
         return vertex.getId().hashCode() + DOT_LABEL_OPENING + getDescriptiveInfo() + DOT_LABEL_CLOSING;
     }
 
+    private Optional<String> getDescriptiveInfoForVertexWithValue() {
+        if (!vertexLabel.isEmpty()) {
+            return Optional.of(vertexLabel);
+        }
+        if (!(vertex instanceof ConstantVertex)) {
```

Comment:
I would prefer `else if` here - it doesn't change the functionality (because of the `return`s but it makes it clearer to the reader that the blocks are mutually exclusive.

(And I know the rest of this file doesn't adhere to this rule, but I would prefer it throughout)